packages/react-native-reanimated/src/errors.ts

```diff
@@ -1,6 +1,23 @@
 'use strict';
 import type { WorkletStackDetails } from './commonTypes';
 
+export interface ReanimatedError extends Error {
+  // eslint-disable-next-line @typescript-eslint/no-misused-new
+  new (message: string): ReanimatedError;
+}
+
+const ReanimatedErrorFactory: ReanimatedError = ((message: string) => {
+  'worklet';
+  const errorInstance = new Error(`[Reanimated] ${message}`);
+  errorInstance.name = 'ReanimatedError';
+  return errorInstance;
+}) as unknown as ReanimatedError;
```

Comment:
We should try something like this, `ErrorConstructor` seems to be the name pattern for such terms.

```tsx
type ReanimatedError = Error & 'ReanimatedError'; // Signed type.

interface ReanimatedErrorConstructor extends Error {
  new (message?: string): ReanimatedError;
  (message?: string): ReanimatedError;
  readonly prototype: ReanimatedError;
}

const ReanimatedError: ReanimatedErrorConstructor = function ReanimatedError(
  message?: string
) {
  // Implementation.
} as ReanimatedErrorConstructor;

const error: ReanimatedError = new Error('ReanimatedError'); // Error due to signed type.

const reaError: Error = new ReanimatedError('ReanimatedError'); // All ok.

const rea2Error: ReanimatedError = new ReanimatedError('ReanimatedError'); // All ok.
```

Reply:
Can't create `const ReanimatedError` variable as it is added to the ignore list on UI and results in being not defined when I use this name.

Reply:
It makes more sense to me too keep the proper name for the type, which shows up when I hover over the `ReanimatedError` call instead of the name of the variable itself, which is assigned to `global.ReanimatedError`, so the name of the variable I initially use can be anything. The `Error` type that is globally available in TS is called `Error`, not `ErrorConstructor`, so I would prefer to follow this convention.

Reply:
And, why do I need to make these strange typing hacks whilst it already works fine?

<img width="573" alt="Screenshot 2024-08-13 at 11 42 06" src="https://github.com/user-attachments/assets/d4940374-5061-47fb-901b-b5d5ebb097bb">